lib/fastlane/plugin/wpmreleasetoolkit/models/firebase_test_runner.rb

```diff
@@ -0,0 +1,88 @@
+require 'json'
+require 'uri'
+require 'fileutils'
+
+module Fastlane
+  class FirebaseTestRunner
+    VALID_TEST_TYPES = %w[instrumentation robo].freeze
+
+    def initialize(key_file:, automatic_login: true)
+      raise "Unable to find key file: #{key_file}" unless File.file? key_file
+
+      @key_file = key_file
+      authenticate if automatic_login
+    end
+
+    def authenticate
+      Action.sh(
+        'gcloud', 'auth', 'activate-service-account',
+        '--key-file', @key_file
+      )
+    end
```

Comment:
💄  To go with the above change suggestion about automatic login behavior being only flagged during init, but only performed later rather than in the `initialize`:
```suggestion
    def authenticate
      Action.sh(
        'gcloud', 'auth', 'activate-service-account',
        '--key-file', @key_file
      )
      @automatic_login = false # We've logged in already now, so no need to redo it again next time
    end
```